### PR TITLE
Make QClass a class, including toString() method

### LINKED ISSUE #7389: `QClass` needs a `toString` method
- Program: Authoritative, Recursor, dnsdist
 - Issue type: Feature request

### Description
In order to log the QClass ('IN', 'CH', 'HS'), it would be nice to have a `toString()` function on the `QClass` object, as well as having an `operator<<(QClass)` for the `Logger` that uses this function.

Once those are there, replacing all the logging of the QClass to use this would be handy as well.

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

And start using it in a few places.
Fixes #7389
### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by omoerbeek** [comment]
I'm not 100% happy yet, pondering if it should be better to have the constants actually be of type QClass instead of the intenral enum type...

**Review by Habbie — CHANGES_REQUESTED** [review]
I like all the code changes that follow from this change (like function arguments having type `QClass`). Some other notes though.

**Comment by Habbie** [comment]
> I'm not 100% happy yet, pondering if it should be better to have the constants actually be of type QClass instead of the internal enum type...

I tried having an opinion on this but I failed.

File: pdns/qtype.cc [hunk 1]
```diff
@@ -160,3 +160,19 @@ QType &QType::operator=(const string &s)
   code = chartocode(s.c_str());
   return *this;
 }
+
+const std::string QClass::toString() const
```
[the remark this answers is not in the document]
Comment (omoerbeek) [reply pdns/qtype.cc:164]: I'd rather go to opposite way and name a `toString` like methods `toString` so I don't have to think what the name is for a method to get a string representation of an object. It really would be nice if all our classes that are potentially loggable should use the same method name, and I think 'getName' is not the right name.

[the remark this answers is not in the document]
Comment (Habbie) [reply pdns/qtype.cc:164]: Right! The point was consistency, I think I also prefer `toString()`, then it's consistent with `DNSName` too. Feel free to leave that rename on QType for a separate PR though.

**Review by Habbie — APPROVED** [review]

**Pull request merged**